src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -211,6 +211,20 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
                 return null;
             }
 
+            // see logic in AddRootCertificate for why this is necessary
+            IntPtr[] issuerHandles = ArrayPool<IntPtr>.Shared.Rent(IntermediateCertificates.Count == 0 ? 1 : IntermediateCertificates.Count);
```

Comment:
`ArrayPool.Rent` might return an array bigger than what you asked for. I don't see the `issuerHandles` getting sliced to the correct size. We just pass it right to `Interop.Crypto.X509DecodeOcspToExpiration`.

That makes me think we might pass a null or garbage handle down depending on what was previously in the array.

Reply:
Something like

```suggestion
            int issuerHandleCount = IntermediateCertificates.Count == 0 ? 1 : IntermediateCertificates.Count
            IntPtr[] issuerHandles = ArrayPool<IntPtr>.Shared.Rent(issuerHandleCount);
```

then pass `issuerHandles.Slice(0, issuerHandleCount)` into X509DecodeToExpiration

Reply:
You could also consider just using `stackalloc` on the handles since pretty much every chain will have an intermediate depth of 32 or less, and just allocate for the insane case.

```C#
Debug.Assert(IntPtr.Size <= 8);
int issuerHandleCount = IntermediateCertificates.Count == 0 ? 1 : IntermediateCertificates.Count;
Span<IntPtr> issuerHandles = issuerHandleCount <= 32 ? stackalloc IntPtr[issuerHandleCount] : new IntPtr[issuerHandleCount];
```

And just skip the array pool entirely.

Reply:
I was going to give a similar suggestion, but this is an `async` method, so it can't have a span as a named local; and it needs to either a) cross the await boundary, or b) do this inside the loop (so after trying each different OCSP endpoint) so it can be factored into a different method.

Reply:
Ah. Whoops. Missed this was in an async context. Scratch that idea then, the pool makes sense.

Reply:
Most recent version does have correct slicing.